joomla/joomla-cms
# [4.0] Multilang associations: Display error when Content Language is trashed or deleted

Pull Request for Issue https://github.com/joomla/joomla-cms/issues/27236

### Summary of Changes
When a Content Language is trashed or deleted
1. Corrected Notice in com_associations Manager
2. Displaying a Warning in each Manager concerned
3. Do not display the concerned associated item in the associations column


### Testing Instructions
Install a multingual site with 3 or more languages using the multilingual sample data.
Trash or delete one of the Content Languages.
Display managers:
articles, categories, menus, contacts, newsfeeds (for contacts and newsfeeds make sure you have associated items including some tagged to the trashed Content Language)
Display Associations component manager after selecting a Type and a Language

-->Screenshots below are captured when the fa-IR content language is trashed.

### Before patch
In managers:
Associated Items tagged to the trashed/deleted Content Language do display.
In Associations component:
We get a Notice. `Notice: Undefined index: link in \layouts\joomla\content\associations.php on line 18`


### After patch
No more Notice.
Associated items which content language is trashed/deleted do not display in the associations column
A Warning is displayed informing the user about the issue.


<img width="1119" alt="Screen Shot 2019-12-23 at 09 56 00" src="https://user-images.githubusercontent.com/869724/71348231-676b4380-256c-11ea-87f8-28c4cede60ab.png">

<img width="1134" alt="Screen Shot 2019-12-23 at 10 09 32" src="https://user-images.githubusercontent.com/869724/71348246-6f2ae800-256c-11ea-9e81-5c4103683ac1.png">

@Quy @Fedik 

Note: PR for the other issue ( `LanguageHelper::getContentLanguages`) discussed in https://github.com/joomla/joomla-cms/issues/27236 is done separately.
